python/pyarrow/conftest.py

```diff
@@ -265,3 +267,19 @@ def add_fs(doctest_namespace, request, tmp_path):
         doctest_namespace["local_path"] = str(tmp_path)
         doctest_namespace["path"] = str(path)
     yield
+
+
+@pytest.fixture(autouse=True)
+def _close_socket(monkeypatch):
+    """
+    There are occasions when a PytestUnraisableExceptionWarning is
+    raised from pyx files or other __del__ impls that doesn't properly
+    call the initial caller's .close when creating sockets. Here we'll
+    ensure sockets are closed.
+    """
+    class socket_(socket.socket):
+        def __del__(self):
+            if not self._closed:
+                self.close()
```

Comment:
We definitely don't want to monkeypatch a core type just to avoid some warnings.

If some of our routines fail to close a socket we should try to improve them. Otherwise we should just live with the warnings.

Reply:
Yeah, this felt dirty. Commit has been dropped, I'll poke around at it a bit more.